### ISSUE TITLE
Pull API Enhancement

### ISSUE THREAD

**Issue by DennisHarper** [ConfigHubPub/ConfigHubPlatform#7]
Hi, my name is Dennis with Belvedere Trading.

How difficult would it be to get the schema information on a repo?  Specifically, we need to get all key Names and Types for a given repo. We would like to be able to verify any configurations that are submitted through the Push API call before they are actually pushed.

Thanks!

**Comment by edoarsla** [comment]
Currently, there is no such *all* request.  There's the INFO API, but that will not give you the keys, just repository information.

Why not make a PULL request (with a context).  You could verify that way.  If that does not cut it, we could expand the INFO API to to accept keys/values flags and respond accordingly.

Does that make sense?

**Comment by DennisHarper** [comment]
Thanks for the reply. 

I cannot make a pull request as the requests are for a given file, which is just template that projects given key:values.  We are trying to get all Keys and their associated types.

**Comment by edoarsla** [comment]
I should have remembered this yesterday.  There's a new API as of version v1.3.3, export, which should do all that you need.  We'll add detailed documentation for it this week.

The request path is `https://confighub-api/rest/info/<account>/<repository>`

It accepts following parameters:
```
@HeaderParam("Repository-Date")
@HeaderParam("Application-Name")
@HeaderParam("Tag")
@HeaderParam("Client-Version")
@HeaderParam("X-Forwarded-For")
@HeaderParam("Pretty")
```

The response you get, is the same format JSON that you would use to PUSH configuration.  The intended idea is to have full repository export, which you can use to import via PUSH to a separate instance.

It supports Tag/Date, so you can export from specific point in time, or if omitted, top-of-tree.

Let me know if this is what you were looking for.

Thanks.
Edin

**Comment by DennisHarper** [comment]
Awesome! I will give it a try

**Comment by DennisHarper** [comment]
Hi Edin,
That works great. The only thing that would be nice is if there was a flag to exclude the values, and just get key metadata. As we enter more and more data, this will probably be a significant amount data coming back, which depending on the use case could be unneeded overhead.

Thanks!
Dennis

**Comment by edoarsla** [comment]
Yes.  We can certainly do that.  We'll add a switch for the values and another for files.  We'll include it in the next build.

**Comment by edoarsla** [comment]
Just released v1.3.5 which includes this:

Expand the Export API to accept additional flags "Omit-Values" and "Omit-Files".
